session/pingpong/consumer_balance_tracker.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package pingpong
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/mysteriumnetwork/node/eventbus"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/identity/registry"
+	"github.com/rs/zerolog/log"
+)
+
+// ConsumerBalanceTracker keeps track of consumer balances
+type ConsumerBalanceTracker struct {
+	balances map[identity.Identity]Balance
+
+	mystSCAddress            common.Address
+	consumerBalanceChecker   consumerBalanceChecker
+	channelAddressCalculator channelAddressCalculator
+	publisher                eventbus.Publisher
+	lock                     sync.Mutex
```

Comment:
Consider putting lock near the field which you want to protect to be more clear.

```go
balances         map[identity.Identity]Balance
balancesLock     sync.Mutex
```